---
title: GWAS of MDD in Generation Scotland, unrelated subsample removing overlap with UK Biobank
author: Mark Adams
output:
  html_document:
    toc: TRUE
    code_folding: "show"
    df_print: kable
---

# Setup

Symlink genotype files into this directory. Set shell variable `GENSCOT` to the mountpoint of the HPC system holding copies of the data for compute:

```{bash, eval=FALSE}

ln -s ${GENSCOT}/genetics/genotypes/GS20K_PLINK_files/QCd_data/QCdGS20K.* .
ln -s ${GENSCOT}/genetics/imputed/HRC/updated_bims/GS20K_HRC_0.8_GCTA_*.* .
ln -s ${GENSCOT}/genetics/imputed/HRC/GS20K_HRC.INFO.gz .
 

```

# Covariates

Combine age/sex covariates with MDS components

```{r}

library(readr)
library(dplyr)
library(stringr)

covars <- read_table2('plinkcovar.txt')

pcs <- read_csv('HM3mds2R.mds.csv')

all_covars <- pcs %>%
inner_join(covars, by=c('FID'='famid', 'IID'='id')) %>%
select(-SOL)

write_delim(all_covars, 'GenScot_AgeSexPCs.cov', delim=' ')

```

# Genotypes

Check sample size with all inclusions and exclusions applied

```{bash, eval=FALSE}

plink \
  --bfile QCdGS20K \
  --pheno QCdGS20K_Pheno_dep_status_Unaff_b_only0_Aff_b_only1.txt \
  --keep QCdGS20K_unrelated_t0.025.fam \
  --remove GSindividualsOverlappingWithUKB500K.txt \
  --freq case-control \
  --maf 0.01 \
  --out GenScot_MDD_rmUKBB \
  --memory 3000

plink \
  --bfile QCdGS20K \
  --pheno QCdGS20K_Pheno_dep_status_Unaff_b_only0_Aff_b_only1.txt \
  --covar GenScot_AgeSexPCs.cov \
  --keep QCdGS20K_unrelated_t0.025.fam \
  --remove GSindividualsOverlappingWithUKB500K.txt \
  --maf 0.01 \
  --logistic \
  --ci 0.95 \
  --out Genscot_MDD_rmUKBB \
  --memory 3000


```

# Imputed

## Case-control allele frequencies

`GSrmUKBB.frq.sh`:

```{bash, eval=FALSE}
#$ -N GSrmUKBB.frq
#$ -l h_rt=1:00:00
#$ -l h_vmem=4G
#$ -t 1-22
#$ -cwd
#$ -e logs
#$ -o logs

. /etc/profile.d/modules.sh
module load igmm/apps/plink/1.90b4

CHR=$SGE_TASK_ID

plink \
  --bfile GS20K_HRC_0.8_GCTA_${CHR} \
  --pheno QCdGS20K_Pheno_dep_status_Unaff_b_only0_Aff_b_only1.txt \
  --keep QCdGS20K_unrelated_t0.025.fam \
  --remove GSindividualsOverlappingWithUKB500K.txt \
  --freq case-control \
  --out GenScot_MDD_rmUKBB.HRC.${CHR} \
  --memory 3000
```

```{bash,eval=FALSE}

qsub GSrmUKBB.frq.sh

```


## Association analysis 

`GSrmUKBB.assoc.sh`:

```{bash, eval=FALSE}
#$ -N GSrmUKBB.assoc
#$ -l h_rt=1:00:00
#$ -l h_vmem=4G
#$ -t 1-22
#$ -cwd
#$ -e logs
#$ -o logs

. /etc/profile.d/modules.sh
module load igmm/apps/plink/1.90b4

CHR=$SGE_TASK_ID

plink \
  --bfile GS20K_HRC_0.8_GCTA_${CHR} \
  --pheno QCdGS20K_Pheno_dep_status_Unaff_b_only0_Aff_b_only1.txt \
  --covar GenScot_AgeSexPCs.cov \
  --keep QCdGS20K_unrelated_t0.025.fam \
  --remove GSindividualsOverlappingWithUKB500K.txt \
  --logistic \
  --ci 0.95 \
  --maf 0.01 \
  --out GenScot_MDD_rmUKBB.HRC.${CHR} \
  --memory 3000
```

```{bash,eval=FALSE}

qsub GSrmUKBB.assoc.sh

```

## Daner file

Open frequency and association results files. Convert to [daner](https://docs.google.com/document/d/1TWIhr8-qpCXB13WCXcU1_HDio8lC_MeWoAg2jlggrtU/edit) format

```{r gs_rmukbb_assoc}

freq <- bind_rows(plyr::llply(1:22, function(chr) read_table2(paste0('GenScot_MDD_rmUKBB.HRC.', chr, '.frq.cc'))))

assoc <- bind_rows(plyr::llply(1:22, function(chr) read_table2(paste0('GenScot_MDD_rmUKBB.HRC.', chr, '.assoc.logistic'))))

```

```{r gs_rmukbb_info}

gs_bim <- read_table2('QCdGS20K.bim', col_names=c('CHR', 'SNP', 'CM', 'BP', 'A1', 'A2'))

hrc_info <- read_table2('GS20K_HRC.INFO.gz')

```

```{r}

# SNPs that are directly genotyped
genotyped <- gs_bim %>%
transmute(CHR, SNP, ngt=1)

sumstats <- 
assoc %>%
filter(TEST == 'ADD') %>%
filter(str_detect(SNP, 'rs')) %>%
left_join(freq, by=c('CHR', 'SNP', 'A1')) %>%
left_join(genotyped, by=c('CHR', 'SNP')) %>%
left_join(hrc_info, by=c('CHR'='CHROM', 'BP'='POS')) %>%
mutate(ngt=if_else(ngt %in% 1, true=1, false=0)) %>%
select(CHR, SNP, BP, A1, A2, FRQ_A_951=MAF_A, FRQ_U_6114=MAF_U,
       INFO, OR, SE, P, ngt) %>%
filter(!is.na(INFO) & FRQ_U_6114 >= 0.01 & FRQ_A_951 > 0.01)

write_tsv(sumstats, 'daner_mdd_genscot_2019-11a_rmUKBB.gz')

```
